File: Selenium/Activities/Activity11.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.wait import WebDriverWait
with webdriver.Firefox() as driver:
    wait = WebDriverWait(driver,timeout=10)
    driver.get("https://training-support.net/webelements/dynamic-controls")
    print(f"The title of the page is: {driver.title}")

    checkBox = driver.find_element(By.ID,"checkbox")
    print("Checkbox is displayed? ", checkBox.is_displayed())
    toggleButton = driver.find_element(By.XPATH,"//button[text()='Toggle Checkbox']")
    toggleButton.click()
    wait.until(EC.invisibility_of_element(checkBox))
    print("Checkbox is displayed? ", checkBox.is_displayed())
    
    toggleButton1 = driver.find_element(By.XPATH,"//button[text()='Toggle Checkbox']")
    toggleButton1.click()
    wait.until(EC.element_to_be_clickable(checkBox)).click()
    print("Checkbox is selected? ", checkBox.is_selected())

    driver.quit()

File: Selenium/Activities/Activity11.java
import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Activity11 {
	public static void main(String args[]) {
		WebDriver driver = new FirefoxDriver();
		WebDriverWait wait = new WebDriverWait(driver,Duration.ofSeconds(10));
		driver.get("https://training-support.net/webelements/dynamic-controls");
		System.out.println("The title of the page is: "+driver.getTitle());
		WebElement checkBox = driver.findElement(By.id("checkbox"));
		System.out.println("Checkbox is visible? " + checkBox.isDisplayed());
		
		WebElement toggleButton = driver.findElement(By.xpath("//button[text()='Toggle Checkbox']"));
		toggleButton.click();
		wait.until(ExpectedConditions.invisibilityOf(checkBox));
		System.out.println("Checkbox is visible? " + checkBox.isDisplayed());
		
		WebElement toggleButton1 = driver.findElement(By.xpath("//button[text()='Toggle Checkbox']"));
		toggleButton1.click();
		wait.until(ExpectedConditions.elementToBeClickable(checkBox)).click();
		System.out.println("Checkbox is visible? " + checkBox.isSelected());
		
		driver.quit();
		
	}

}
